src/status_im/contexts/wallet/send/select_collectible_amount/view.cljs

```diff
@@ -46,24 +43,23 @@
        :container-style style/network-tags-container}]
 
      [quo/amount-input
-      {:max-value       (if (integer? balance) balance 0)
-       :min-value       1
-       :value           value
-       :on-inc-press    inc-value
-       :on-dec-press    dec-value
+      {:max-value       (controlled-input/upper-limit value)
+       :min-value       (controlled-input/lower-limit value)
+       :value           (controlled-input/numeric-value value)
+       :on-inc-press    (fn [] (set-value #(controlled-input/increase %)))
+       :on-dec-press    (fn [] (set-value #(controlled-input/decrease %)))
        :container-style style/amount-input-container
        :status          (if incorrect-value? :error :default)}]
      [quo/bottom-actions
       {:actions          :one-action
        :button-one-props {:on-press  #(rf/dispatch
                                        [:wallet/set-collectible-amount-to-send
                                         {:stack-id :screen/wallet.select-collectible-amount
-                                         :amount   value}])
+                                         :amount   (controlled-input/numeric-value value)}])
                           :disabled? incorrect-value?}
        :button-one-label (i18n/label :t/confirm)}]
      [quo/numbered-keyboard
       {:left-action :none
        :delete-key? true
-       :on-press    add-digit
-       :on-delete   delete-digit}]]))
-
+       :on-press    (fn [c] (set-value #(controlled-input/add-character % c)))
+       :on-delete   (fn [] (set-value controlled-input/delete-last))}]]))
```

Comment:
To avoid not-needed re-renders, consider passing stable references to these callbacks too, please.